core/arch/arm/plat-stm/rng_support.c

```diff
@@ -89,8 +89,8 @@ TEE_Result hw_get_random_bytes(void *buf, size_t len)
 
 	static int nbcall;  /* debug purpose - 0 is the initial value*/
 
-	volatile uint32_t tmpval[_LOCAL_FIFO_SIZE/2];
-	int i;
+	volatile uint32_t tmpval[_LOCAL_FIFO_SIZE / 2] = { 0 };
```

Comment:
That should be `= { };`, and the checkpatch warning makes sense: why is `volatile` needed here? What kind of unwanted optimization is it supposed to avoid? (this is more a question for @etienne-lms, although this is 8 year old code :wink:)

Reply:
I think it's not needed at all :)